internal/slicer/slicer.go

```diff
@@ -169,35 +163,33 @@ func Run(options *RunOptions) (*Report, error) {
 		if o.Mode.IsDir() {
 			relPath = relPath + "/"
 		}
-		listed := false
-		until := untilPaths[relPath]
+		inSliceContents := false
+		until := setup.UntilMutate
 		for _, extractInfo := range extractInfos {
 			if extractInfo.Context == nil {
 				continue
 			}
-			s, ok := extractInfo.Context.(*setup.Slice)
+			slice, ok := extractInfo.Context.(*setup.Slice)
 			if !ok {
-				panic(fmt.Errorf("internal error: invalid Context of type %T in extractInfo", extractInfo.Context))
+				return fmt.Errorf("internal error: invalid Context of type %T in extractInfo", extractInfo.Context)
 			}
-			pathInfo, ok := s.Contents[extractInfo.Path]
+			pathInfo, ok := slice.Contents[extractInfo.Path]
 			if !ok {
-				panic(fmt.Errorf("internal error: path %q not listed in slice", extractInfo.Path))
+				return fmt.Errorf("internal error: path %q not listed in slice contents", extractInfo.Path)
 			}
-			listed = true
+			inSliceContents = true
 
-			if pathInfo.Until == setup.UntilMutate && until == unset {
-				until = mutate
-			} else if pathInfo.Until == setup.UntilNone {
-				until = keep
+			if pathInfo.Until == setup.UntilNone {
+				until = setup.UntilNone
```

Comment:
For the record, this looks a bit different from the suggested.  Not a big deal for now as the result is the same, but this version doesn't work once we have more than two options for this value.